projects/core/src/site-context/config/config-loader/site-context-config-initializer.ts

```diff
@@ -1,44 +1,131 @@
+import { Injectable } from '@angular/core';
+import { ConfigInitializer } from 'projects/core/src/config';
+import { Observable } from 'rxjs';
+import { filter, map, take } from 'rxjs/operators';
+import { BaseSite } from '../../../model/misc.model';
+import { JavaRegExpConverter } from '../../../util/java-reg-exp-converter/java-reg-exp-converter';
+import { WindowRef } from '../../../window/window-ref';
+import { BaseSiteService } from '../../facade/base-site.service';
 import {
-  ConfigInitializer,
-  CONFIG_INITIALIZER,
-} from '../../../config/config-initializer/config-initializer';
-import { FeatureConfigService } from '../../../features-config/services/feature-config.service';
-import { SiteContextConfigLoaderService } from './../../config/config-loader/site-context-config-loader.service';
-import { SiteContextConfig } from './../../config/site-context-config';
-import { BASE_SITE_CONTEXT_ID } from './../../providers/context-ids';
-
-/**
- * Initializes the site context config
- */
-export function initSiteContextConfig(
-  configLoader: SiteContextConfigLoaderService,
-  config: SiteContextConfig,
-  featureConfigService: FeatureConfigService
-): ConfigInitializer | null {
-  // TODO(#11515): in 4.0 drop the feature level guard
-  if (!featureConfigService.isLevel('3.2')) {
-    return null;
+  BASE_SITE_CONTEXT_ID,
+  CURRENCY_CONTEXT_ID,
+  LANGUAGE_CONTEXT_ID,
+  THEME_CONTEXT_ID,
+} from '../../providers/context-ids';
+import { SiteContextConfig } from '../site-context-config';
+
+@Injectable({ providedIn: 'root' })
+export class SiteContextConfigInitializer implements ConfigInitializer {
+  readonly scopes = ['context'];
+  readonly configFactory = () => this.resolveConfig().toPromise();
```

Comment:
configFactory function should return `Promise.resolve({})` when:
`!featureConfigService.isLevel('3.2') || config?.context?.[BASE_SITE_CONTEXT_ID]` is truthy

Otherwise it can return `this.resolveConfig().toPromise();`